test/runtests.jl

```diff
@@ -3634,6 +3634,15 @@ end
     @test NoPrimal(ReverseSplitWithPrimal) === ReverseSplitNoPrimal
 end
 
+@testset "needs_primal" begin
+    @test needs_primal(Reverse) === false
```

Comment:
Tests fail here, presumably this should be moved into the EnzymeCore tests [not enzyme proper tests]

Reply:
I can do that, but I wrote it here because other parts of EnzymeCore are actually tested here and not in EnzymeCore's tests (e.g. `WithPrimal`).

Also, it should also be solved by exporting `needs_primal`.

Reply:
yeah we didn't use to have CI for EnzymeCore separately, but since we now do, might as well put in the right place (and ideally move the rest)